core/src/main/java/org/apache/iceberg/deletes/DVFileWriter.java

```diff
@@ -36,6 +36,14 @@ public interface DVFileWriter extends Closeable {
    */
   void delete(String path, long pos, PartitionSpec spec, StructLike partition);
 
+  default void delete(
```

Comment:
JavaDoc, also worth explaining the alternative which is calling delete for every position.....but I think it's worth introducing an API on the writer which takes an in memory index and writes out all the deleted positions.